fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1420,808 @@ def draw_act(self, obj, layout, value):
+
+        # jump to custom level script array
+        match_line, sig_insert_line, default_line = self.find_export_lines(
+            script_lines,
+            match_str=f"JUMP_LINK({fast64_level_script})",
+            fast64_signature="JUMP_LINK(",
+            alt_condition="",
+            start_delim="ALLOC_LEVEL_POOL(",
+            end_delim="AREA(",
+        )
+
+        if not match_line and sig_insert_line:
+            script_lines.insert(sig_insert_line + 1, f"\tJUMP_LINK({fast64_level_script}),\n")
+        elif not match_line and default_line:
+            script_lines.insert(default_line, f"\tJUMP_LINK({fast64_level_script}),\n")
+
+        self.write_file_lines(script_path, script_lines)
+
+    def export_behavior_header(self, context, props):
+        # check if behavior_header.h exists
+        decomp_path = Path(bpy.path.abspath(bpy.context.scene.decompPath))
+        behavior_header = decomp_path / "include" / "behavior_data.h"
+        if not behavior_header.exists():
+            PluginError("could not find behavior_data.h")
+
+        bhv_header_lines = open(behavior_header, "r").readlines()
+        export_bhv_include = f"extern const BehaviorScript {props.bhv_name}[];\n"
+        fast64_sig = "// fast64 object exports get inserted here"
+
+        match_line, sig_insert_line, default_line = self.find_export_lines(
+            bhv_header_lines,
+            match_str=export_bhv_include,
+            fast64_signature=fast64_sig,
+            alt_condition='#include "types.h"',
+        )
+
+        if match_line:
+            bhv_header_lines[match_line] = export_bhv_include
+        elif sig_insert_line:
+            bhv_header_lines.insert(sig_insert_line + 1, export_bhv_include)
+        else:
+            export_line = default_line + 1 if default_line else len(bhv_header_lines)
+            bhv_header_lines.insert(export_line, f"\n{fast64_sig}\n")
+            bhv_header_lines.insert(export_line + 1, export_bhv_include)
+
+        self.write_file_lines(behavior_header, bhv_header_lines)
+
+    # export the behavior script, edits /data/behaviour_data.c and /include/behaviour_data.h
+    def export_behavior_script(self, context, props):
+        # make sure you have a bhv script
+        if len(props.behavior_script) == 0:
+            raise PluginError("Behavior must have more than 0 cmds to export")
+
+        # export the behavior script itself
+        decomp_path = Path(bpy.path.abspath(bpy.context.scene.decompPath))
+        behavior_data = decomp_path / "data" / "behavior_data.c"
+        if not behavior_data.exists():
+            PluginError("could not find behavior_data.c")
+
+        # add at top of bhvs, 3 lines after this is found
+        bhv_data_lines = open(behavior_data, "r").readlines()
+        export_bhv_name = f"const BehaviorScript {props.bhv_name}[] = {{\n"
+        last_bhv_define = "#define SPAWN_WATER_DROPLET(dropletParams)"
+        fast64_sig = "// fast64 object exports get inserted here"
```

Comment:
There is no benefit code wise as far as I know, and I feel like it would be more likely to cause issues if you only did parsing inside a block instead of the whole file. If for example you want to replace an existing model name, and only parse in a block, you would just redefine the model instead of searching the entire file, or insert something outside the block which may confuse users.